packages/nimble-components/src/rich-text-mention/users/tests/rich-text-mention-users.spec.ts

```diff
@@ -0,0 +1,316 @@
+        mappingUser.displayName = mapping.displayName ?? '';
+        newUserMappingElements.push(mappingUser);
+    });
+    element.replaceChildren(...newUserMappingElements);
+    await waitForUpdatesAsync();
+}
+
+describe('RichTextMentionUsers', () => {
+    let element: RichTextMentionUsers;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+
+    // prettier-ignore
+    async function setup(
+        mappings: BasicUserMentionMapping[],
+        pattern = ''
+    ): Promise<Fixture<RichTextMentionUsers>> {
+        return fixture<RichTextMentionUsers>(html`
+            <${richTextMentionUsersTag} pattern="${pattern}">
+                ${repeat(() => mappings, html<BasicUserMentionMapping>`
+                    <${mappingUserTag}
+                        key="${x => x.key}"
+                        display-name="${x => x.displayName}">
+                    </${mappingUserTag}>
+                `)}
+            </${richTextMentionUsersTag}>`);
+    }
+
+    afterEach(async () => {
+        if (disconnect) {
+            await disconnect();
+        }
+    });
+
+    it('should export its tag', () => {
+        expect(richTextMentionUsersTag).toBe('nimble-rich-text-mention-users');
+    });
+
+    it('can construct an element instance', () => {
+        expect(
+            document.createElement('nimble-rich-text-mention-users')
+        ).toBeInstanceOf(RichTextMentionUsers);
+    });
+
+    it('should have character, icon and pattern in mention internals', async () => {
+        ({ element, connect, disconnect } = await setup(
+            [
+                { key: 'user:1', displayName: 'user' },
+                { key: 'user:2', displayName: 'user' }
+            ],
+            'user:'
+        ));
+        await connect();
+        expect(element.mentionInternals.character).toBe('@');
+        expect(element.mentionInternals.icon).toBe(iconAtTag);
+        expect(element.mentionInternals.pattern).toBe('user:');
+    });
+
+    it('should have mappingConfigs based on mapping elements', async () => {
+        const mappingData = [
+            { key: 'user:1', displayName: 'user' },
+            { key: 'user:2', displayName: 'user' }
+        ];
+        ({ element, connect, disconnect } = await setup(mappingData, 'user:'));
+        await connect();
+        const mappingConfigs = element.mentionInternals.mentionConfig!.mappingConfigs;
+        expect(mappingConfigs?.size).toEqual(2);
+        expect(Array.from(mappingConfigs.keys())).toEqual(['user:1', 'user:2']);
+        expect(mappingConfigs.get('user:1')?.displayName).toEqual('user');
+        expect(mappingConfigs.get('user:2')?.displayName).toEqual('user');
+        expect(mappingConfigs.get('user:1')?.mentionHref).toEqual('user:1');
+        expect(mappingConfigs.get('user:2')?.mentionHref).toEqual('user:2');
+    });
+
+    it('should update mappingConfigs when mapping elements changed', async () => {
```

Comment:
we should have tests for when observable properties change. Such as display-name and mention-href updating (both from valid to valid and valid to invalid)